src/hooks/useCheckedItem.ts

```diff
@@ -0,0 +1,75 @@
+import { useState } from "react";
+
+const useCheckedItem = (length: number) => {
+  const [isChecked, setIsChecked] = useState<boolean[]>(
+    new Array(length).fill(false)
+  );
+  const [isDeleted, setIsDeleted] = useState<boolean[]>(
+    new Array(length).fill(false)
+  );
+  const [isCheckedAll, setIsCheckedAll] = useState<boolean>(false);
```

Comment:
요게 필요할까요? 그냥 변수로 설정해도 될 값인 것 같아서요!
```suggestion
  const isCheckedAll = isChecked.every(checked => checked);
```

Reply:
현재 체크로직을 전반적으로 수정하여 check된 아이템 id 배열을 갖고 있는 형식으로 되어있습니다. 그에 맞춰 전체적으로 조금씩 로직이 바뀌었으니 한번 확인 부탁드리겠습니다!